packages/react/src/rise.tsx

```diff
@@ -56,6 +56,14 @@ export type ActionModelState<
   $: 'action'
   name: T
 } & K
+export type ActionModelStatePayload<T> = Omit<T, 'name' | '$'>
+export type ActionDefinition<
+  T extends ActionModelState<any, any>,
```

Comment:
why are there two generic arguments to ActionDefinition? seems like you need either T or K not both

also it is best practice to have a good name for the generic. the single letter thing is common but not very good for people trying to understand the code

Reply:
Second argument `K` is provided with default value so that I don't have to write:
```
export type ActionDefinition<
  T extends ActionModelState<any, any>
> = {
  action: (payload: ActionModelStatePayload<T>) => void
  validate?: (args: unknown) => ActionModelStatePayload<T>
}
```
but can move `ActionModelStatePayload<T>` into "variable".

Reply:
> also it is best practice to have a good name for the generic. the single letter thing is common but not very good for people trying to understand the code

Yeah, I am in support for this change. I would say - maybe - makes sense to do a follow-up where we rename all Ts and Ks in the project to something more descriptive. Good opportunity to improve naming in general throughout the codebase (it's mostly just `rise.tsx` file where types live, so not that much of work)

Reply:
fine to do it in a followup!

and yes please lets just use T

Reply:
Fine! Applied - I think less letters, the better lol!